cms/djangoapps/contentstore/rest_api/v2/views/downstreams.py

```diff
@@ -111,6 +128,58 @@ class _AuthenticatedRequest(Request):
 #         ...
 
 
+@view_auth_classes()
+class UpstreamListView(DeveloperErrorViewMixin, APIView):
+    """
+    Serves course->library publishable entity links
+    """
+    def get(self, request: _AuthenticatedRequest, course_key_string: str):
+        """
+        Fetches publishable entity links for given course key
+        """
+        try:
+            course_key = CourseKey.from_string(course_key_string)
+        except InvalidKeyError as exc:
+            raise ValidationError(detail=f"Malformed course key: {course_key_string}") from exc
+        links = PublishableEntityLink.get_by_downstream_context(downstream_context_key=course_key)
+        serializer = PublishableEntityLinksSerializer(links, many=True)
+        return Response(serializer.data)
+
+
+@view_auth_classes()
+class DownstreamContextListView(DeveloperErrorViewMixin, APIView):
+    """
+    Serves library block->courses links
+    """
+    def get(self, _: _AuthenticatedRequest, usage_key_string: str) -> Response:
+        """
+        Fetches downstream context links for given publishable entity
+        """
+        try:
+            usage_key = UsageKey.from_string(usage_key_string)
+            print(usage_key)
+        except InvalidKeyError as exc:
+            raise ValidationError(detail=f"Malformed usage key: {usage_key_string}") from exc
+        links = PublishableEntityLink.get_by_upstream_usage_key(upstream_usage_key=usage_key)
+        downstream_key_list = [link.downstream_context_key for link in links]
+
+        # Count the number of times each course is linked to the library block
+        counter = Counter(downstream_key_list)
+
+        result = []
+        for context_key, count in counter.most_common():
+            # The following code only can handle the correct display_name for Courses as context
+            course = modulestore().get_course(context_key)
```

Comment:
We can search for course names using [CourseOverview](https://github.com/openedx/edx-platform/blob/88e9b08c6b8458b221a2a969aa7bf46f1efb2a80/openedx/core/djangoapps/content/course_overviews/models.py#L45) table. Something like below:

```python
CourseOverview.objecs.filter(id__in=context_key_list).values_list('display_name', 'id')
```